src/main/java/analyzer/analyzerInterfaces/AnalyzerComment.java

```diff
@@ -0,0 +1,22 @@
+package analyzer.analyzerInterfaces;
+
+import java.util.ArrayList;
+import java.util.Optional;
+
+public interface AnalyzerComment {
+
+    // Constructed message with all the template variables applied.
+    String message();
+
+    // The message with template variables in there.
+    String template();
+
+    // The identifier for the template on website-copy.
+    String externalTemplate();
+
+    // Not sure if this should be a map
+    ArrayList<String> variables();
```

Comment:
It should be a map, see https://exercism.org/docs/building/tooling/analyzers/interface#h-output-format for an example of the output format. I'd also suggest naming it `params` to match the field in the specification (unless there is a good reason not to of course).

Reply:
Ok understood. Working on this now. @ErikSchierboom

Reply:
The key/value pairs map placeholder keys to values (see https://github.com/exercism/website-copy/blob/main/analyzer-comments/csharp/gigasecond/use_1e9_not_math_pow.md for a placeholder example), which are then applied before returning the actual analyzer comment.